lib/msal-node/src/network/HttpClient.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { INetworkModule, NetworkRequestOptions } from '@azure/msal-common';
+import { HttpMethod } from './../utils/Constants';
+import axios from 'axios';
+
+/**
+ * This class implements the Fetch API for GET and POST requests. See more here: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API
+ */
+export class HttpClient implements INetworkModule {
+    /**
+     * Axios library for REST endpoints - Get request
+     * @param url
+     * @param headers
+     * @param body
+     */
+    async sendGetRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<T> {
+        // setHeaders
+        this.setHeaders(options);
+
+        // axios config
+        const config = {
+            method: HttpMethod.GET,
+            url: url
+        };
+
+        // GET call
+        const response = await axios(config);
+        return await response as unknown as T;
+    }
+
+    /**
+     * Axios Client for REST endpoints - Post request
+     * @param url
+     * @param headers
+     * @param body
+     */
+    async sendPostRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<T> {
+        // set Headers
+        this.setHeaders(options);
+
+        // axios config
+        const config = {
+            method: HttpMethod.POST,
+            url: url,
+            body: (options && options.body) || ''
+        };
+
+        const response = await axios(config);
+        return await response as unknown as T;
+    }
+
+    private setHeaders(options?: NetworkRequestOptions): void {
```

Comment:
I think that this is not what we want to do here, This is for setting defaults on requests, but each request is unique and I would worry we are not cleaning up headers. 

Axios config accepts headers as a key 

```js
const request = {
    method: HttpMethod.POST,
    url, 
    headers: options.headers
}

```

Reply:
yes I am aware. But options.Headers is a `Map<string, string>` - defined by msal-common already. I agree defaults may not be a good idea, considering we change telemetry option, I think the best approach here is to change it while populating headers in msal-common, which I plan to in a follow up PR.

Reply:
We should not be setting defaults at every request. What's the problem with options.Headers being a Map<string, string> ?

This is currently working for me in the device code implementation: 

```
 private async executePostRequest<T>(url: string, headers: Map<string, string>, body: string): Promise<T>{

        const response =  await axios({
            method: 'post',
            url: url,
            data: body,
            headers: headers,
          });

          return response.data as T;
```

Reply:
Awesome! Interesting,  this did not compile for me, may be my vscode is throwing me.  Will clean and do it again, I found issues with linking for me.  Will compile and check.

Reply:
Done! It is my local environment, I was able to compile and test now.